File: Android/app/src/main/java/com/example/mpass/model/tmdbModels.kt
package com.example.mpass.model

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.example.mpass.api.GenreVideos
import com.example.mpass.api.Genres
import com.example.mpass.api.TmdbApi
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch

class TmdbModels {
    private val tmdbApi = TmdbApi()

    fun fetchTvGenres(): LiveData<Genres> {
        val data = MutableLiveData<Genres>()
        GlobalScope.launch {
            tmdbApi.tvGetGenre()?.let {
                data.postValue(it)
            }
        }
        return data
    }

    fun fetchTvByGenre(genreId: Int): LiveData<GenreVideos> {
        val data = MutableLiveData<GenreVideos>()
        GlobalScope.launch {
            tmdbApi.tvGetGenreCollection(genreId)?.let {
                data.postValue(it)
            }
        }
        return data
    }
}

File: iOS/mpass/model/tmdbModels.swift
import Foundation
import PromiseKit
import SwiftyJSON

class TmdbModels {
    let api = TmdbApi()
    
    func fetchTvGenres() -> LiveData<JSON> {
        let data = LiveData<JSON>()
        firstly {
            api.tvGetGenre()
        }.done { genres in
            data.data = genres
        }.catch { err in
            data.error = err
        }
        return data
    }
    
    func fetchTvByGenre(_ gid: Int) -> LiveData<JSON> {
        let data = LiveData<JSON>()
        firstly {
            api.tvGetGenreCollection(gid)
        }.done { tvs in
            var json = tvs
            json["gid"] = JSON(gid)
            data.data = json
        }.catch { err in
            data.error = err
        }
        return data
    }
}

